src/components/pages/dashboard-pages/EditLesson.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { getLessonDetails, editLesson } from '../../../util/DataBaseRequests';
+import { useAuth } from '../../../AuthProvider';
+import EditLessonForm from './EditLessonForm';
+
+const EditLesson = () => {
+  const navigate = useNavigate();
+  const { userData } = useAuth();
+  const location = useLocation();
+  const { selectedStudentId, lessonId } = location.state || {};
+
+  const [formData, setFormData] = useState({
+    lessonTitle: '',
+    classId: '',
+    lessonDescription: '',
+    type: '',
+    hometask: '',
+    lessonSchedule: { date: '', startTime: '' },
+  });
+  const [formErrors, setFormErrors] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchClasses = async () => {
+      const token = userData.token;
+      try {
+        const response = await getLessonDetails(token, selectedStudentId, lessonId);
+        const lessonInfo = response.data.lesson;
+
+        setFormData({
+          lessonTitle: lessonInfo.lessonTitle || '',
+          classId: lessonInfo.classId || '',
+          lessonDescription: lessonInfo.lessonDescription || '',
+          type: lessonInfo.type || '',
+          hometask: lessonInfo.hometask || '',
+          lessonSchedule: {
+            date: formatDate(lessonInfo.lessonSchedule?.date),
+            startTime: lessonInfo.lessonSchedule?.startTime || '',
+          },
+        });
+      } catch (error) {
+        console.error('Error fetching lesson:', error);
```

Comment:
I think the user should be informed if something went wrong

Reply:
Fixed